multiboot2-header/src/header.rs

```diff
@@ -59,6 +61,28 @@ impl<'a> Multiboot2Header<'a> {
         Self { inner: reference }
     }
 
+    /// Find the header in a given slice.
+    pub fn find_header(buffer: &[u8]) -> Option<(&[u8], u32)> {
+        // the magic is 32 bit aligned and inside the first 8192 bytes
+        assert!(buffer.len() >= 8192);
+        let mut chunks = buffer[0..8192].chunks_exact(4);
+        let magic_index = match chunks.position(|vals| {
+            u32::from_le_bytes(vals.try_into().unwrap()) // yes, there's 4 bytes here
+            == MULTIBOOT2_HEADER_MAGIC
+        }) {
+            Some(idx) => idx * 4,
+            None => return None,
+        };
+        chunks.next(); // arch
+        let header_length: usize = u32::from_le_bytes(chunks.next().unwrap().try_into().unwrap())
```

Comment:
> A lot of unwraps in one line. I think, this could could be written a little nicer. Perhaps, returning a Result would be better?

The last unwrap shouldn't fail (it just converts 4 bytes to an `u32`), but the first one might, so yeah, returning an error for a header that is cut off might be sensible. What would be an appropriate error type here?

Reply:
I think an error similar to <https://github.com/rust-osdev/multiboot2/blob/main/multiboot2/src/lib.rs#L165> would be nice

Reply:
I could think of something like `-> Result<Option<Header>, BufError>`.

BufError could be something like `::ToSmall`, `::Unaligned` or so